service/service_setup.go

```diff
@@ -45,7 +45,7 @@ func (svc *serviceCore) setupLogging() {
 		svc.log = logBuilder.WithConfiguration(svc.logConfig).Build()
 		ulog.SetLogger(svc.log)
 	} else {
-		svc.log.Debug("Using custom log provider due to service.WithLogger option")
+		ulog.Logger(context.Background()).Debug("Using custom log provider due to service.WithLogger option")
```

Comment:
We have a service.WithLogger option during the service setup (similar to WithModule, etc). If that is set, we don't update the service.Logger